tests/e2e/vmservice_vm.go

```diff
@@ -0,0 +1,176 @@
+)
+
+var _ bool = ginkgo.Describe("[vmsvc] vm service with csi vol tests", func() {
+
+	f := framework.NewDefaultFramework("vmsvc")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	f.SkipNamespaceCreation = true // tests will create their own namespaces
+	var (
+		client            clientset.Interface
+		namespace         string
+		datastoreURL      string
+		storagePolicyName string
+		storageClassName  string
+		storageProfileId  string
+		vcRestSessionId   string
+		vmi               string
+		vmopC             ctlrclient.Client
+		cnsopC            ctlrclient.Client
+	)
+
+	ginkgo.BeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		client = f.ClientSet
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		bootstrap()
+		vcRestSessionId = createVcSession4RestApis()
+
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		storageClassName = strings.ReplaceAll(storagePolicyName, "_", "-") // since this is a wcp setup
+
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		dsRef := getDsMoRefFromURL(ctx, datastoreURL)
+		framework.Logf("dsmoId: %v", dsRef.Value)
+
+		storageProfileId = e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+		contentLibId := createAndOrGetContentlibId4Url(vcRestSessionId, GetAndExpectStringEnvVar(envContentLibraryUrl),
+			dsRef.Value, GetAndExpectStringEnvVar(envContentLibraryUrlSslThumbprint))
+
+		namespace = createTestWcpNs(
+			vcRestSessionId, storageProfileId, vmClassBestEffortSmall, contentLibId, getSvcId(vcRestSessionId))
+
+		vmopScheme := runtime.NewScheme()
+		gomega.Expect(vmopv1.AddToScheme(vmopScheme)).Should(gomega.Succeed())
+		vmopC, err = ctlrclient.New(f.ClientConfig(), ctlrclient.Options{Scheme: vmopScheme})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		cnsOpScheme := runtime.NewScheme()
+		gomega.Expect(cnsop.AddToScheme(cnsOpScheme)).Should(gomega.Succeed())
+		cnsopC, err = ctlrclient.New(f.ClientConfig(), ctlrclient.Options{Scheme: cnsOpScheme})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		vmImageName := GetAndExpectStringEnvVar(envVmsvcVmImageName)
+		framework.Logf("Waiting for virtual machine image list to be available in namespace '%s' for image '%s'",
+			namespace, vmImageName)
+		vmi = waitNGetVmiForImageName(ctx, vmopC, namespace, vmImageName)
+		gomega.Expect(vmi).NotTo(gomega.BeEmpty())
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		dumpSvcNsEventsOnTestFailure(client, namespace)
+		delTestWcpNs(vcRestSessionId, namespace)
+		waitForNamespaceToGetDeleted(ctx, client, namespace, poll, pollTimeout)
+	})
+
+	ginkgo.It("basic test", func() {
```

Comment:
we should come up valid testcase name . change the test case name to meaningful sentence